packages/dashboard/src/managers/robot-trajectory-manager.ts

```diff
@@ -62,37 +68,30 @@ interface Request {
 interface Response {
   response: string;
   values: unknown;
+  error?: string;
 }
 
-export class DefaultTrajectoryManager {
-  static async create(url: string): Promise<DefaultTrajectoryManager> {
-    let ws = new WebSocket(url);
-    await new Promise<void>((res, rej) => {
-      ws.addEventListener('open', function listener() {
-        ws.removeEventListener('open', listener);
-        res();
-      });
-
-      ws.addEventListener('error', function listener(e) {
-        ws.removeEventListener('error', listener);
-        rej(e);
-      });
-    });
-    return new DefaultTrajectoryManager(ws);
+export class DefaultTrajectoryManager extends TrajectorySocketManager {
+  constructor(ws: WebSocket, authenticator?: Authenticator) {
+    super();
+    if (ws) this._webSocket = ws;
+    if (authenticator) this._authenticator = authenticator;
   }
 
   async latestTrajectory(request: TrajectoryRequest): Promise<TrajectoryResponse> {
-    const event = await this._send(JSON.stringify(request));
+    if (appConfig.isKeycloak) await this._authenticator?.refreshToken();
```

Comment:
It should refresh the token regardless of the authenticator, the stub authenticator can just be a no-op.